### ISSUE TITLE
BBDownloader has regressed somehow: doesn't check sizes of files and skip loading them

### ISSUE THREAD

**Issue by rjmcdougall** [rjmcdougall/BB#85]
For some reason, BBdownloader will re-download files even if it has them after the directory has changed. 

This means a simple profile change will cause a complete re-download.

**Comment by danielkeithw** [comment]
I'll check

**Comment by danielkeithw** [comment]
A few things to check:

1) Google Drive will not provide size on HTTP Header.
2) Possible Null Values
3) Changes where videos do/not have lengths now that everything can have a length.

**Comment by danielkeithw** [comment]
Did locate a bug where Mp3 in some circumstance were getting NaN lengths.  Length doesn't matter for this algorithm, though.

**Comment by danielkeithw** [comment]
I'm going to remove GetURLFileSize.  Google Cloud doesn't support it because that metadata is missing.

**Comment by danielkeithw** [comment]
I've tested dozens of permutations of this and cannot reproduce the behavior you see.  I've tried your test profiles, etc.  My only explanation is an old version of the code OR a slight behavior change.  See #3.  Note this is not downloading new files, just announcing that ANY change has been found.

The approach I switched to in a previous commit:

1) Download the Directory file
2) check the size of the new and old download JSON objects
3) if the file has changed announce that changes have been found. this could be algorithm reorders or new/deleted files. This does NOT cause new files to be downloaded.
4) Check for existence/equality of all mp3/mp4/m4a files.  Equality determined by the Size attribute on the new and old JSON.  It is NOT parsed from disk.
5) If any files have changed, download them.
6) complete, clean up.
